### ISSUE TITLE
Interface compilation ambiguities

### ISSUE THREAD

**Issue by MJCWilhelm** [amusecode/amuse#587]
With a recent update, the way interfaces are compiled has changed. Before, it was done through 'make clean/make' in the directory of an interface inside src/amuse/community, but that method no longer works (giving an error in build.py, where it can't find the AMUSE module). Currently, it is done through 'make {code}.code' in the main AMUSE directory. 

This results in an ambiguity when making new interfaces. Specifically, the makefile generated by the build.py script in dir mode uses the amusifier executable to compile an interface, instead of build.py. This does still work with the old make method, but not with the new method. In all new interface makefiles, the line 'CODE_GENERATOR = $(AMUSE_DIR)/build.py' has been added, which means build.py will be used for compilation instead of amusifier. This does work with the new method, but not the old. 

It would be most straightforward to change the generated makefile to include the new definition and state in the tutorial that the new interface should be generated in the community folder, but I can also see that one would want to develop an interface in another place than the amuse directory. In either case, the interface tutorials should be edited to reflect this ambiguity.

Any thoughts?

**Comment by ipelupessy** [comment]
hmm, the make inside the code directory should still work (i do this all the time)..specifically it should always be possible to make a makefile which works in any location (like you say) with a make in the dir.
you may need to set some environment variables though...can you give an example of a code that does not work? (it may also be the makefile) and the error? (and how is amuse installed: I assume a straight clone of the repo with ./configure and make framework done?)

build.py should be the same as amusifier (except that amusifier lifes in the bin directory, you should set a PATH for it to work)..build.py is often found by some relative AMUSE_DIR specification (this is unreliable) in the makefiles

**Comment by MJCWilhelm** [comment]
I tried this on both sse and ph4, the error I get is:

```
/data1/wilhelm/py_envs/amuse_dev/amuse//build.py --type=c interface.py ph4Interface -o worker_code.cc
Traceback (most recent call last):
  File "/data1/wilhelm/py_envs/amuse_dev/amuse//build.py", line 7, in <module>
    from amuse.rfi import gencode
ModuleNotFoundError: No module named 'amuse'
make: *** [Makefile:61: worker_code.cc] Error 1
```
The installation is indeed done through ./configure and make (but not 'make framework' I think?). 

Also, this is on a strw pc, where I need to use 'mpiexec $AMUSE_DIR/amuse.sh script.py' to use amuse. However, using 'mpiexec make' gives the same error, and 'mpiexec $AMUSE_DIR/amuse.sh make' results in '/data1/wilhelm/py_envs/amuse_dev/amuse_env/bin/python: can't open file 'make': [Errno 2] No such file or directory'.

**Comment by ipelupessy** [comment]
ok, since you don't install, you need to set PYTHONPATH and a PATH for everything to work 
```
export PYTHONPATH=/home/inti/code/amuse/amuse/src:${PYTHONPATH}
export PATH=/home/inti/code/amuse/amuse/bin:${PATH}
```
note that you can also (and this is slightly preferred) use a virtualenv and a develop pip install (just for completeness I give all the steps):
```
virtualenv env
source env/bin/activate
git clone git@github.com:amusecode/amuse
cd amuse
pip install -e .
```
then everything should work (and you don't need amuse.sh)...except that it doesn't ofcourse..

**Comment by ipelupessy** [comment]
...I am just trying this out...

**Comment by ipelupessy** [comment]
ok, I fixed the build up in #588, can you try and report?

**Comment by ipelupessy** [comment]
will be solved by #588
